struqture/src/spins/spin_noise_operator.rs

```diff
@@ -321,17 +351,23 @@ impl SpinLindbladNoiseOperator {
         right: &DecoherenceOperator,
         value: CalculatorComplex,
     ) -> Result<(), StruqtureError> {
+        if left.is_empty() || right.is_empty() {
+            return Err(StruqtureError::InvalidLindbladTerms);
+        }
+
         for ((decoherence_product_left, value_left), (decoherence_product_right, value_right)) in
             left.iter().cartesian_product(right.iter())
         {
-            let value_complex = value_right.conj() * value_left;
-            self.add_operator_product(
-                (
-                    decoherence_product_left.clone(),
-                    decoherence_product_right.clone(),
-                ),
-                value_complex * value.clone(),
-            )?;
+            if !decoherence_product_left.is_empty() && !decoherence_product_right.is_empty() {
```

Comment:
I think check here necessary? Shouldn't this be addressed in the `add_...` function?

Reply:
`add_operator_product` returns an Err if an identity is passed, and this Err would be propagated to the returned value of `add_noise_from_full_operators`. We don't want this, because we just want to avoid adding terms that contain identities, which could normally appear during the computation but would cancel out.